server/src/main/java/org/candlepin/dto/rules/v1/ActivationKeyDTO.java

```diff
@@ -0,0 +1,415 @@
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import org.apache.commons.lang.builder.EqualsBuilder;
+import org.apache.commons.lang.builder.HashCodeBuilder;
+import org.candlepin.dto.CandlepinDTO;
+import org.candlepin.jackson.CandlepinAttributeDeserializer;
+import org.candlepin.jackson.CandlepinLegacyAttributeSerializer;
+import org.candlepin.util.MapView;
+import org.candlepin.util.SetView;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * A DTO representation of the ActivationKey entity in the Rules framework:
+ * <tt>
+ *  {
+ *    "id": "string",
+ *    "pools": [
+ *      {
+ *        "quantity": 0,
+ *        "pool": {
+ *          "id": "string",
+ *          "attributes": [ "key1": "value1", "key1": "value2"],
+ *          "productAttributes": [ "key1": "value1", "key1": "value2"]
+ *        }
+ *      }
+ *    ]
+ *  }
+ * </tt>
+ */
+public class ActivationKeyDTO extends CandlepinDTO<ActivationKeyDTO> {
+    public static final long serialVersionUID = 1L;
+
+    /**
+     * Join object DTO for joining the activation key to pools
+     */
+    public static class ActivationKeyPoolDTO extends CandlepinDTO<ActivationKeyPoolDTO> {
+        protected InternalPoolDTO pool;
+        protected Long quantity;
+
+        public ActivationKeyPoolDTO() {
+            // Intentionally left empty
+        }
+
+        public InternalPoolDTO getPool() {
+            return this.pool;
+        }
+
+        public ActivationKeyPoolDTO setPool(InternalPoolDTO pool) {
+            this.pool = pool;
+            return this;
+        }
+
+        public Long getQuantity() {
+            return this.quantity;
+        }
+
+        public ActivationKeyPoolDTO setQuantity(Long quantity) {
+            this.quantity = quantity;
+            return this;
+        }
+
+        @Override
+        public boolean equals(Object obj) {
+            if (obj == this) {
+                return true;
+            }
+
+            if (obj instanceof ActivationKeyPoolDTO) {
+                ActivationKeyPoolDTO that = (ActivationKeyPoolDTO) obj;
+
+                EqualsBuilder builder = new EqualsBuilder()
+                    .append(this.getPool(), that.getPool())
+                    .append(this.getQuantity(), that.getQuantity());
+
+                return builder.isEquals();
+            }
+
+            return false;
+        }
+
+        @Override
+        public int hashCode() {
+            HashCodeBuilder builder = new HashCodeBuilder(37, 7)
+                .append(this.getPool())
+                .append(this.getQuantity());
+
+            return builder.toHashCode();
+        }
+    }
+
+    /**
+     * Internal DTO that represents a Pool object on an ActivationKeyPool object.
+     */
+    public static class InternalPoolDTO extends CandlepinDTO<InternalPoolDTO> {
```

Comment:
Not sure how much of a fan I am of this class. I assume the intent here is to reduce the amount of data we have to serialize, but it starts to get away from the standardization the DTOs are trying to fix. I'll need to think on this a bit.

Reply:
Yes, the intent is to minimize the data, since there already is a PoolDTO for rules, but it has way more fields than we actually need (the existing PoolDTO has 12 fields, while in this case we only use 3 of them). I also don't like the way this is written, and struggled but didn't find a better way.

Reply:
Since the rules are effectively on their deathbed anyway, we can go ahead and merge this as-is, I suppose. Since we're in need of a rebase anyway, could you add a note or some such on it to indicate that it's a non-standard solution to this problem and one we shouldn't get in the habit of doing?

Reply:
Done